cpp/daal/src/algorithms/low_order_moments/oneapi/low_order_moments_distributed_oneapi_impl.i

```diff
@@ -0,0 +1,488 @@
+    for (uint32_t distrBlockId = 0; distrBlockId < nDistrBlocks; distrBlockId++)
+    {
+        PartialResult * inputPartialResult = static_cast<PartialResult *>((*partResultsCollection)[distrBlockId].get());
+        BlockDescriptor<algorithmFPType> blockDesc;
+
+        NumericTablePtr tablePtr = inputPartialResult->get((PartialResultId)nObservations);
+        DAAL_CHECK_STATUS_VAR(tablePtr ? tablePtr->getBlockOfRows(0, 1, readOnly, blockDesc) : services::Status(ErrorNullPartialResult));
+        algorithmFPType * pBlockObsCount = blockDesc.getBlockPtr();
+        DAAL_CHECK_MALLOC(pBlockObsCount);
+
+        auto bNVecHost = bNVec.template get<algorithmFPType>().toHost(ReadWriteMode::writeOnly, status);
+        DAAL_CHECK_STATUS_VAR(status);
+        bNVecHost.get()[distrBlockId] = *pBlockObsCount;
+
+        *pNObservations += *pBlockObsCount;
+        tablePtr->releaseBlockOfRows(blockDesc);
+    }
+
+    size_t rowSize = nFeatures * sizeof(algorithmFPType);
+
+    for (uint32_t i = 0; i < TaskInfoDistributed<algorithmFPType, scope>::nPartialResults; i++)
+    {
+        // copy partial results from each block into one common buffer
+        DAAL_ASSERT(bAuxHostBuffers[i].size() == nDistrBlocks * nElemsInStride);
+        for (uint32_t distrBlockId = 0; distrBlockId < nDistrBlocks; distrBlockId++)
+        {
+            PartialResult * inputPartialResult = static_cast<PartialResult *>((*partResultsCollection)[distrBlockId].get());
+            BlockDescriptor<algorithmFPType> blockDesc;
+            NumericTablePtr tablePtr = inputPartialResult->get((PartialResultId)TaskInfoDistributed<algorithmFPType, scope>::resPartialIds[i]);
+            DAAL_CHECK_STATUS_VAR(tablePtr ? tablePtr->getBlockOfRows(0, 1, readOnly, blockDesc) : services::Status(ErrorNullPartialResult));
+            DAAL_ASSERT(blockDesc.getBuffer().size() == nFeatures);
+            int result = daal::services::internal::daal_memcpy_s(bAuxHostBuffers[i].get() + distrBlockId * nElemsInStride, rowSize,
+                                                                 blockDesc.getBlockPtr(), rowSize);
+            DAAL_CHECK(!result, services::ErrorMemoryCopyFailedInternal);
+            tablePtr->releaseBlockOfRows(blockDesc);
+        }
+        DAAL_ASSERT_UNIVERSAL_BUFFER(bAuxBuffers[i], algorithmFPType, nDistrBlocks * nElemsInStride);
+        context.copy(bAuxBuffers[i], 0, (void *)bAuxHostBuffers[i].get(), nDistrBlocks * nElemsInStride, 0, nDistrBlocks * nElemsInStride, status);
+        DAAL_CHECK_STATUS_VAR(status);
+    }
+
+    /* merge blocks */
+    auto kMergeDistrBlocks = factory.getKernel(TaskInfoDistributed<algorithmFPType, scope>::kMergeDistrBlocksName, status);
+    DAAL_CHECK_STATUS_VAR(status);
+    {
+        KernelRange range(nFeatures);
+
+        KernelArguments args(3 + TaskInfoDistributed<algorithmFPType, scope>::nPartialResults + TaskInfoDistributed<algorithmFPType, scope>::nBuffers
+                                 + 1 /*rows in block info*/,
```

Comment:
```suggestion
        KernelArguments args(4 + TaskInfoDistributed<algorithmFPType, scope>::nPartialResults + TaskInfoDistributed<algorithmFPType, scope>::nBuffers,
```

Reply:
+1 /*rows in block info*/ was added to have similar look and fill with batch and online kernels where it is added as 
 + (TaskInfoBatch<algorithmFPType, scope>::isRowsInBlockInfoRequired ? 1 : 0)